### DISCUSSION TITLE
Nゴルフ　プロジェクト#1

### DISCUSSION THREAD

**Discussion by kenton116** [kenton116/n-development-team#10] [Show and tell]
ブラウザゲーム作りのためのディスカッション

**Comment by fuuchin** [comment]
slackにゲームの設計の写真を載せときました！！

**Reply by kenton116** [reply]
ありがとうございます

**Comment by kenton116** [comment]
みなさん、平面ゴルフ（仮称）プロジェクト始めてもよろしいでしょうか？
![509EE0B3-350A-43E1-8211-70F39B0C0818](https://user-images.githubusercontent.com/57275744/145661437-75496f81-4a8f-4e11-98d9-d2c40ee8c388.jpeg)

**Reply by fuuchin** [reply]
おｋです

**Reply by fuuchin** [reply]
僕はPhotoShopもイラストレーターもないので画像系のはkenton116さんにお願いしたいです、

**Reply by ihwka** [reply]
すいません、なかなか顔出せず。

面白そうですね！
確かに「野田ゲーっぽく」なら2Dでカジュアルな感じがいいですよね。
それに3Dになると手も足も出そうにないですしw

時間的にも技術的にもどこまで協力できるかわからないですが
内容的にはいいと思います。賛成です。

**Reply by satsukizzz** [reply]
とりあえず、素案から見えてくる必要作業の列挙します
ロジック部分
点数の計算
鳥やバンカーやカップなどの当たり判定幅
ターン制の処理（対戦時）
課金ボタンに応じた当たり判定幅の増加
モグラや鳥が出る頻度
「ショットがランダムに……」の処理
OBの点数
玉の弾道計算


UI部分
打球するのはキーを押すのか画面内にボタン設置してあってそれを押すのか
課金時のカップの大きさ変化
（もっとある気がする、デザインとの分け方よくわからず）



デザイン
メインキャラ、その動くエフェクト、喜びや悲しみ
ボールが飛ぶエフェクト
打球ゲージ
もぐら
鳥
課金ボタン
待っている二人目（対戦時）
さらに…
画面の大きさ（ピクセル）と比
canvas使うかなどフレームワークの問題


音
BGM、打音、OB（ファー？）、砂、モグラの妨害音、風の音、ギャラリーの歓声、カップに入る音、ランク（点数）に応じた効果音、鳥の鳴き声、メインキャラの喜びや悲しみ

**Comment by fuuchin** [comment]
皆さんもどうですかーー

**Comment by fuuchin** [comment]
ゲーム名はNゴルフとかでよろしいでしょうか？

**Comment by kenton116** [comment]
初期コミットしました！
HTML系はぼくが担当させていただきます。

ゴルフ原型作成係、１人モード作成係、２人モード作成係、デザイン係
それぞれどれを担当したいかお願いします

**Reply by kenton116** [reply]
作成の順番としては、HTML、ゴルフ原型、１人モード、２人モード、デザインです。

**Reply by fuuchin** [reply]
できるかどうか厳しいかもですが、ゴルフ原型行きたいと思います

**Reply by kenton116** [reply]
OKです！
HTMLコミット（自分ができるまでお待ちください）

**Comment by ihwka** [comment]
この時点でもう何をどう作っていくのかわからなくて
自分が何をできるのかわからないんですが

**Reply by kenton116** [reply]
えっと、できそうなところで気楽に参加してみてください。わからなかったら質問してください！

**Reply by satsukizzz** [reply]
画像や音の素材集めが並行してできることかもしれないです

**Comment by satsukizzz** [comment]
現時点で、トップページから1人プレイ、2人プレイに行くという遷移がないので、そういったところは作ってもいいでしょうか。
そこもkentonさんかな

**Comment by satsukizzz** [comment]
やることリストアップして分担していくのがいいかな？？？ウーン

**Comment by kenton116** [comment]
HTMLは組んでおきました。
ふーちんさんにはn-golf/fuuchinという名前でブランチ切ってください🙇‍♂️
